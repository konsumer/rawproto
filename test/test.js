import { readFileSync } from 'fs'
import hexy from 'hexy'
import { getData, getProto } from '../index.js'
import protobuf from 'protobufjs'

// build a demo message
const proto = await protobuf.load(new URL('demo.proto', import.meta.url).pathname)

const Test = proto.lookupType('Test')
const msg = Test.create({
  nums: [1, 2, 3, 4, 5],
  num: 1,
  str: 'hello',
  children: [
    { num: 1, str: 'cool', children: [{ num: 1 }] },
    { num: 2, str: 'awesome', children: [{ num: 2 }] },
    { num: 3, str: 'neat', children: [{ num: 3 }] }
  ]
})
const encoded = Test.encode(msg).finish()

console.log('Protobuf:')
console.log(hexy.hexy(encoded))

console.log('Structure:')
console.log(JSON.stringify(getData(encoded), null, 2), '\n')

console.log('Structure (all binary):')
console.log(JSON.stringify(getData(encoded, 'binary'), null, 2), '\n')

console.log('Proto:')
console.log(getProto(encoded))

console.log('Proto (all binary):')
console.log(getProto(encoded, 'binary'))
