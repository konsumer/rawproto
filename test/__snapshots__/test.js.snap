// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rawproto Some missing fields 1`] = `
Array [
  Object {
    "1": Object {
      "data": Array [
        1,
        2,
        3,
        4,
        5,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "2": 1,
  },
  Object {
    "3": "hello",
  },
  Object {
    "4": Object {
      "data": Array [
        8,
        1,
        18,
        4,
        99,
        111,
        111,
        108,
        26,
        2,
        8,
        1,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "4": Object {
      "data": Array [
        8,
        2,
        18,
        7,
        97,
        119,
        101,
        115,
        111,
        109,
        101,
        26,
        2,
        8,
        2,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "4": Object {
      "data": Array [
        8,
        3,
        18,
        4,
        110,
        101,
        97,
        116,
        26,
        2,
        8,
        3,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "5": "this one is extra",
  },
]
`;

exports[`rawproto Test raw 1`] = `
Array [
  Object {
    "1": Object {
      "data": Array [
        1,
        2,
        3,
        4,
        5,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "2": 1,
  },
  Object {
    "3": "hello",
  },
  Object {
    "4": Object {
      "data": Array [
        8,
        1,
        18,
        4,
        99,
        111,
        111,
        108,
        26,
        2,
        8,
        1,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "4": Object {
      "data": Array [
        8,
        2,
        18,
        7,
        97,
        119,
        101,
        115,
        111,
        109,
        101,
        26,
        2,
        8,
        2,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "4": Object {
      "data": Array [
        8,
        3,
        18,
        4,
        110,
        101,
        97,
        116,
        26,
        2,
        8,
        3,
      ],
      "type": "Buffer",
    },
  },
]
`;

exports[`rawproto Test2 raw 1`] = `
Array [
  Object {
    "1": Object {
      "data": Array [
        1,
        2,
        3,
        4,
        5,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "2": 1,
  },
  Object {
    "3": "hello",
  },
  Object {
    "4": Object {
      "data": Array [
        8,
        1,
        18,
        4,
        99,
        111,
        111,
        108,
        26,
        2,
        8,
        1,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "4": Object {
      "data": Array [
        8,
        2,
        18,
        7,
        97,
        119,
        101,
        115,
        111,
        109,
        101,
        26,
        2,
        8,
        2,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "4": Object {
      "data": Array [
        8,
        3,
        18,
        4,
        110,
        101,
        97,
        116,
        26,
        2,
        8,
        3,
      ],
      "type": "Buffer",
    },
  },
  Object {
    "5": "this one is extra",
  },
]
`;

exports[`rawproto getProto 1`] = `
"syntax = \\"proto3\\";

message MessageRoot {
  bytes field1 = 1; // could be a repeated-value, string, bytes, or malformed sub-message
  int32 field2 = 2; // could be a int32, int64, uint32, bool, enum, etc, or even a float of some kind
  string field3 = 3; // could be a repeated-value, string, bytes, or malformed sub-message
  bytes field4 = 4; // could be a repeated-value, string, bytes, or malformed sub-message
  bytes field4 = 4; // could be a repeated-value, string, bytes, or malformed sub-message
  bytes field4 = 4; // could be a repeated-value, string, bytes, or malformed sub-message
  string field5 = 5; // could be a repeated-value, string, bytes, or malformed sub-message
}"
`;
